File: Programming Basics/Nested Loops - Exercises/03.SumPrimeNonPrime.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        string input = Console.ReadLine();

        int primeSum = 0, nonPrimeSum = 0;

        while (input != "stop")
        {
            int number = int.Parse(input);

            if (number < 0)
            {
                Console.WriteLine("Number is negative.");
            }
            else if (IsPrime(number))
            {
                primeSum += number;
            }
            else
            {
                nonPrimeSum += number;
            }

            input = Console.ReadLine();
        }

        Console.WriteLine($"Sum of all prime numbers is: {primeSum}");
        Console.WriteLine($"Sum of all non prime numbers is: {nonPrimeSum}");
    }
    public static bool IsPrime (int number)
    {
        int n;
        for (n = 2; n <= number - 1; n++)
        {
            if (number % n == 0)
            {
                return false;
            }
        }
        if (number == n)
        {
            return true;
        }
        return false;
    } 
}

File: Programming Basics - Python/Nested Loops - Exercises/03.SumPrimeNonPrime.py
prime_sum = 0
non_prime_sum = 0

while True:
    entry = input()

    if entry == "stop":
        print(f"Sum of all prime numbers is: {prime_sum}")
        print(f"Sum of all non prime numbers is: {non_prime_sum}")
        exit()

    number = int(entry)

    if number < 0:
        print("Number is negative.")
        continue

    if number <= 1:
        non_prime_sum += number
        continue

    is_prime = True
    for i in range(2, number):
        if number % i == 0:
            is_prime = False
            break

    if is_prime:
        prime_sum += number
    else:
        non_prime_sum += number